File: Task7.1P/app/src/main/java/com/example/lostandfoundapp/db/AppDatabase.java
package com.example.lostandfoundapp.db;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.example.lostandfoundapp.model.Advert;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

@Database(entities = {Advert.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    public abstract AdvertDao advertDao();

    private static volatile AppDatabase INSTANCE;
    private static final int NUMBER_OF_THREADS = 4;
    public static final ExecutorService databaseWriteExecutor =
            Executors.newFixedThreadPool(NUMBER_OF_THREADS);

    public static AppDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (AppDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                                    AppDatabase.class, "lost_found_db_java")
                            .build();
                }
            }
        }
        return INSTANCE;
    }
}

File: AndroidStudioProjects/TaskManager/app/src/main/java/com/example/taskmanager/db/AppDatabase.kt
package com.example.taskmanager.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.taskmanager.model.Task

@Database(entities = [Task::class], version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun taskDao(): TaskDao

    companion object {
        @Volatile
        private var INSTANCE: AppDatabase? = null

        fun getDatabase(context: Context): AppDatabase {
            return INSTANCE ?: synchronized(this) {
                val instance = Room.databaseBuilder(
                    context.applicationContext,
                    AppDatabase::class.java,
                    "task_manager_db" // Name of the SQLite database file
                )
                    // In a production app, you'd handle migrations here
                    .build()
                INSTANCE = instance
                instance
            }
        }
    }
}